jobs/ltv_daily.py

```diff
@@ -0,0 +1,206 @@
+    search_rfm_ds = search_rfm_full.limit(TRAINING_SAMPLE).select(columns).toPandas()
+    for metric in pred_metrics:
+        print(metric)
+        # train and extract model performace
+        model_perf, model = train_metric(search_rfm_ds, metric, plot=False, penalty=0.8)
+        model_perf["pct"] = model_perf.Model / (model_perf.Actual + 1) - 1
+        model_perf["metric"] = metric
+        model_perf["date"] = submission_date
+        model_perf_data = pd.concat([model_perf_data, model_perf])
+
+        # make predictions using model
+        @F.udf(DoubleType())
+        def ltv_predict_metric(metric, model=model):
+            import lifetimes
+
+            return ltv_predict(
+                PREDICTION_DAYS, metric.frequency, metric.recency, metric.T, model
+            )
+
+        # go back to full sample here
+        predictions = search_rfm_full.select(
+            "*", ltv_predict_metric(metric).alias("prediction_" + metric)
+        )
+
+        if not model_pred_data:
+            model_pred_data = predictions
+        else:
+            model_pred_data = model_pred_data.join(
+                predictions.select("client_id", "prediction_" + metric), on="client_id"
+            )
+
+    predictions = F.create_map(
+        list(
+            chain(
+                *((F.lit(name), F.col("prediction_" + name)) for name in pred_metrics)
+            )
+        )
+    ).alias("predictions")
+
+    model_pred_data = model_pred_data.withColumn("predictions", predictions)
+    model_perf_data["active_days"] = model_perf_data.index
+    model_perf_data_sdf = spark.createDataFrame(model_perf_data)
+
+    (
+        model_perf_data_sdf.write.format("bigquery")
+        .option("table", f"{project_id}.{dataset_id}.{model_input_table_id}")
+        .option("temporaryGcsBucket", temporary_gcs_bucket)
+        .mode("overwrite")
+        .save()
+    )
+
+    (
+        model_pred_data.write.format("bigquery")
+        .option("table", f"{project_id}.{dataset_id}.{model_output_table_id}")
+        .option("temporaryGcsBucket", temporary_gcs_bucket)
+        .mode("overwrite")
+        .save()
+    )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(__doc__)
+    parser.add_argument("--submission-date", default="2020-03-03")
+    parser.add_argument("--project-id", default="moz-fx-data-bq-data-science")
+    parser.add_argument(
+        "--source-qualified-table-id",
+        default="moz-fx-data-shared-prod.search.search_rfm",
+    )
+    parser.add_argument("--dataset-id", default="bmiroglio")
+    parser.add_argument("--intermediate-table-id", default="search_rfm_day")
+    parser.add_argument("--model-input-table-id", default="ltv_daily_model_perf_script")
+    parser.add_argument("--model-output-table-id", default="ltv_daily_script")
+    parser.add_argument(
+        "--temporary-gcs-bucket", default="moz-fx-data-bq-data-science-bmiroglio"
+    )
+    args = parser.parse_args()
+
+    main(**vars(args))
```

Comment:
@benmiroglio Do you might taking a copy of this script and running it locally, just to verify that I havent functionally changed anything in the course of writing this?

Reply:
I'll try to give a whirl now.